src/devices_models/device_constructors/regulationdevice_constructor.jl

```diff
@@ -11,7 +11,7 @@ function construct_device!(
         throw(ArgumentError("AGC is only compatible with AreaBalancePowerModel"))
     end
 
-    devices = get_available_components(model.device_type, sys)
+    devices = get_available_components(model.component_type, sys)
```

Comment:
do we have getter functions for these fields? If we do please use the getters or make some. Since these are functions that potential users would re-implement then we should always try to make them use getters.